src/app/component/profile/profile.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { AuthenticationService } from 'src/app/service/authentication.service';
+import { ModalDismissReasons, NgbModal } from '@ng-bootstrap/ng-bootstrap';
+
+@Component({
+  selector: 'app-profile',
+  templateUrl: './profile.component.html',
+  styleUrls: ['./profile.component.scss'],
+})
+export class ProfileComponent implements OnInit, OnDestroy {
+  dataSource: any;
+  userData: any;
+  closeResult = '';
```

Comment:
why handling form like this ?

Reply:
I have Remove dataSource and close result